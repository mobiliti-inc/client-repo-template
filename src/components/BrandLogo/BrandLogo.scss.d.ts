// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'brand': string;
  'brand__name': string;
  'fades-in-out': string;
}
declare var cssExports: CssExports;
export = cssExports;
