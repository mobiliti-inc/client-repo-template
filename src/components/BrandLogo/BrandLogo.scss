@import '../../styles/variables';

.brand {
    display: flex;
    flex-direction: column;
    align-items: center;
    color: $white;
    &__name {
        display: flex;
        font-family: $body-font;
        font-style: normal;
        font-weight: normal;
        font-size: 1.3em;
        line-height: 36px;
    }
}
