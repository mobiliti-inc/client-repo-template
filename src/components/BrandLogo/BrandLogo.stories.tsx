import { storiesOf } from "@storybook/react";
import { text, object } from "@storybook/addon-knobs";
import React from "react";
import { BrandLogo } from "../";

storiesOf("BrandLogo", module)
	.add("Default", () =>
	<BrandLogo
		brandName={text('brandName', "")}
		style={ object('style', { background: 'rgba(73, 150, 174, 0.8)' })}
		logo={text('logo', "")}
	/>);
