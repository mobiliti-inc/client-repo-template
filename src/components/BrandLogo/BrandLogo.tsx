import React, { FC, ReactNode } from "react";
import { brandLogo } from "../../assets";

import "./BrandLogo.scss";

type customStyle = {
	background?: string;
	color?: string;
	height?: string;
	width?: string;
};

interface IProps {
	style?: customStyle;
	logo?: string;
	brandName?: ReactNode;
}

const BrandLogo: FC<IProps> = ({ style, logo, brandName }) => (
		<div style={style} styleName="brand">
			<img src={logo || brandLogo} alt="Brand Logo" />
			<span styleName="brand__name">{brandName || "carsales.com"}</span>
		</div>
	);

export default BrandLogo;
