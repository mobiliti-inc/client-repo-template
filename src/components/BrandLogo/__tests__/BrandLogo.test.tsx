import React from 'react';

import BrandLogo from '../BrandLogo';

describe('BrandLogo', () => {
	it('renders', () => {
		const subject = shallow(<BrandLogo />);
		expect(subject).toMatchSnapshot();
	});

	it('renders with brand Name', () => {
		const brandNameText = "Mobiliti.com";
		const subject = shallow(<BrandLogo brandName={brandNameText} />);
		expect(subject).toMatchSnapshot();

		const brandNameSpan =subject.find('.brand__name');
		expect(brandNameSpan).toHaveLength(1);
		expect(brandNameSpan.props().children).toEqual(brandNameText);
	});

	it('renders with style', () => {
		const customStyle = {'background': 'red'};
		const subject = shallow(<BrandLogo style={customStyle} />);
		expect(subject).toMatchSnapshot();
		const brandDiv =subject.find('.brand');
		expect(brandDiv).toHaveLength(1);
		expect(brandDiv.props().style.background).toEqual(customStyle.background);
	});

	it('renders with logo', () => {
		const subject = shallow(<BrandLogo logo='logo.png' />);
		expect(subject).toMatchSnapshot();
		const brandImg =subject.find('img');
		expect(brandImg).toHaveLength(1);
		expect(brandImg.props().src).toEqual('logo.png');
	});
});
