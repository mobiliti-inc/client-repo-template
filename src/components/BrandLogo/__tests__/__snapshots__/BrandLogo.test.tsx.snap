// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo renders 1`] = `
<div
  className="brand"
>
  <img
    alt="Brand Logo"
    src="test-file-stub"
  />
  <span
    className="brand__name"
  >
    carsales.com
  </span>
</div>
`;

exports[`BrandLogo renders with brand Name 1`] = `
<div
  className="brand"
>
  <img
    alt="Brand Logo"
    src="test-file-stub"
  />
  <span
    className="brand__name"
  >
    Mobiliti.com
  </span>
</div>
`;

exports[`BrandLogo renders with logo 1`] = `
<div
  className="brand"
>
  <img
    alt="Brand Logo"
    src="logo.png"
  />
  <span
    className="brand__name"
  >
    carsales.com
  </span>
</div>
`;

exports[`BrandLogo renders with style 1`] = `
<div
  className="brand"
  style={
    Object {
      "background": "red",
    }
  }
>
  <img
    alt="Brand Logo"
    src="test-file-stub"
  />
  <span
    className="brand__name"
  >
    carsales.com
  </span>
</div>
`;
